### PR TITLE
Add standardised color names to doc and Color constructor accepting t…

### PR DESCRIPTION
**Pull request by Nutriz**
For #11164, added color names to the doc.

I discovered the `@GDScript.ColorN()` with this issue (but thinked many times : why Color hasn't a constructor with named color ?), so I added a constructor to do this, same as `@GDScript.ColorN()`.

### CODE REVIEW

**Review by bojidar-bg — APPROVED** [review]

**Comment by bojidar-bg** [comment]
Would be nice if we get code-completion as well, but, it isn't blocking.

**Comment by Jerome67000** [comment]
Ho yes, I will try to add it :)

**Comment by akien-mga** [comment]
Well if this constructor works (I thought it wasn't possible? But maybe I just extended to Color/Color8 case to ColorN), by all means let's drop `ColorN`. We don't need two methods doing the same thing.

File: core/variant_call.cpp [hunk 1]
```diff
@@ -887,6 +887,10 @@ struct _VariantCall {
 		r_ret = Color::hex(*p_args[0]);
 	}
 
+	static void Color_init5(Variant &r_ret, const Variant **p_args) {
+		r_ret = Color::named(*p_args[0], *p_args[1]);
+	}
+
```
Comment (akien-mga) [inline core/variant_call.cpp]: Style issue here (trailing tab).

**Comment by poke1024** [comment]
Also see https://github.com/godotengine/godot/pull/14704

**Comment by Jerome67000** [comment]
Yes @akien-mga, my first goal was to move this from `@GDScript.ColorN()` to `Color()`, but after, I have thinked retro compatibility. If isn't a big deal, I can remove `@GDScript.ColorN()`.

You thought it wasn't possible because `Color` class already had a constructor expecting a String (for html colors), the trick is to add float alpha to the prototype (same as `@GDScript.ColorN()` ).

Another solution can be to remove support of html color without `#` at start, and having one unique constructor for both html colors and named color, with only a String, no alpha for second arg.

**Comment by akien-mga** [comment]
> Another solution can be to remove support of html color without # at start, and having one unique constructor for both html colors and named color, with only a String, no alpha for second arg.

I'd be fine with that personally. Or if others prefer to keep alpha, it could then work as `Color("#330e12", 0.4)` and `Color("darkred", 0.7)`, but we'd have to decide what should happen for `Color("#330e12aa", 0.1)`... multiply `aa` and `0.1` to get the final alpha value?

**Comment by poke1024** [comment]
This clearly conflicts with https://github.com/godotengine/godot/pull/14704 (meaning, it will have to be removed if https://github.com/godotengine/godot/pull/14704 ever comes). Also as an API breaking change, how can this be 3.0 at this point? @akien-mga Want a clear explanation, please.

**Comment by akien-mga** [comment]
@poke1024 The proposal here is a well delimited change that would remove the awkward `ColorN` GDScript method to the profit of a cleaner `Color(String name)` constructor. It's indeed a bit late for 3.0 in theory, but my personal opinion is that if we're going to deprecate `ColorN`, it's best to do it before 3.0 is released, so that we don't need to drag a deprecated duplicated method for all of 3.x's life cycle.

Your PR #14704 achieved something similar, but with a much bigger scope redesigning part of GDScript and with a much higher risk for regression, so at this stage it's too late for 3.0 and will have to wait for 3.1 if it can be implemented in a way that does not break 3.0.x projects.

I can understand if you feel that it's unfair to your work on #14704, and if so I'm fine moving this PR to 3.1 too and we can assess then which of the two has the most interesting implementation. And we'll have to live with ColorN for a couple years more, but it's not a big deal and we'll always find stuff we'd like to remove before 3.0 goes stable, so maybe it's better to draw the line now.

**Comment by poke1024** [comment]
I'm fine with having this as a bridging solution for 3.0, but I'd find it unfair if this gets merged for 3.0 and then #14704 gets killed in 3.1 because of it.

**Comment by akien-mga** [comment]
Yes, I agree I should draw a stronger line. It's a bit tough to have to make judgment calls on hundreds of issues and dozens of PRs every day to get closer to 3.0 for real, depending on where I am in my triage streak I change opinions :)

Thanks for your recent focus on fixing critical issues btw, it's very welcome.

**Comment by akien-mga** [comment]
Note that another PR to just add the missing documentation would be welcome for 3.0.

**Comment by Jerome67000** [comment]
Well, I did not see your PR @poke1024, it wasn't tagged in #11164 issue, nice work ! Like akien-mga said, bigger scope = higher risk for regression, I understand that was moved to 3.1. My PR as very limited scope and can be nice to remove `ColorN()` before 3.0.

I propose to change my PR to remove `ColorN()`, having a `Color()` for named colors, not implementing autocompletion (only completing doc), and use it as a bridge, waiting your nicer solution with #14704. What do you thing ?

**Comment by poke1024** [comment]
@Jerome67000 The color name docs are definitely a nice 3.0 addition I guess, I missed them too. As for `ColorN` and the `Color` constructor, I think you should keep a version of this PR for 3.1. After all my solution had some reservations. I just think we should find one new solution for named colors for 3.1 and then stick with it long term.

**Comment by poke1024** [comment]
@Jerome67000 To expand on this. If you find a way to add context sensitive code-completion to your solution, that would be pretty cool and indeed make #14704 obsolete. I just don't see how this would work without having to go a very similar path. It would need an autocompletion that basically knows it's in the new Color constructor version taking a String and then present the list of colors.

**Comment by Jerome67000** [comment]
Alright, I will create another PR for adding only the names to the doc, removing it from this PR and we will thinks togethers to have a nice solution for 3.1 !

**Comment by akien-mga** [comment]
Superseded by #14704.

**Pull request closed without merging**